docs/web-hooks.md

```diff
@@ -0,0 +1,69 @@
+# Web hooks
+
+⚠ This feature is experimental!
+
+## Configuration
+
+The plugin "web-hooks" needs to be installed and loaded for this feature to work.
+
+You can trigger an HTTP POST request to a URL when a Xen Orchestra API method is called.
+
+* Go to Settings > Plugins > Web hooks
+* Add new hooks
+* For each hook, configure:
+  * Method: the XO API method that will trigger the HTTP request when called
+  * Type:
+    * pre: the request will be sent when the method is called
+    * post: the request will be sent after the method action is completed
+  * URL: the full URL which the requests will be sent to
+* Save the plugin configuration
+
+From now on, a request will be sent to the corresponding URLs when a configured method is called by an XO client.
+
+## Request content
+
+```
+POST / HTTP/1.1
+Content-Type: application/json
+```
+
+The request's body is a JSON string representing an object with the following properties:
+
+- `type`: `"pre"` or `"post"`
+- `callId`: unique ID for this call to help match a pre-call and a post-call
+- `userId`: unique internal ID of the user who performed the call
+- `userName`: login/e-mail address of the user who performed the call
+- `method`: name of the method that was called (e.g. `"vm.start"`)
+- `params`: call parameters (object)
+- `result`: call result on success ("post" hooks only)
+- `error`: call result on error ("post" hooks only)
```

Comment:
What do think of adding a timestamp for post and pre, and the duration for post?